src/spec-common/injectHeadless.ts

```diff
@@ -512,38 +512,41 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 					name: progressName,
 					status: 'succeeded',
 				});
+			} catch (err) {
+				infoOutput.event({
+					type: 'progress',
+					name: progressName,
+					status: 'failed',
+				});
+				if (err?.cmdOutput) {
+					infoOutput.raw(`\r\n\x1b[1m${err.cmdOutput}\x1b[0m\r\n\r\n`);
+				}
+				if (err && (err.code === 130 || err.signal === 2)) { // SIGINT seen on darwin as code === 130, would also make sense as signal === 2.
+					infoOutput.raw(`\r\n\x1b[1m${lifecycleHookName} interrupted.\x1b[0m\r\n\r\n`);
+				} else {
+					if (err?.code) {
+						infoOutput.write(toErrorText(`${lifecycleHookName} failed with exit code ${err.code}. Skipping any further user-provided commands.`));
+					}
+					throw new ContainerError({
+						description: `The ${lifecycleHookName} in the ${userCommandOrigin} failed.`,
+						originalError: err
+					});
+				}
 			}
+		}
 
-			infoOutput.raw(`\x1b[1mRunning the ${lifecycleHookName} from ${userCommandOrigin}...\x1b[0m\r\n\r\n`);
+		infoOutput.raw(`\x1b[1mRunning the ${lifecycleHookName} from ${userCommandOrigin}...\x1b[0m\r\n\r\n`);
 
-			let commands;
-			if (typeof userCommand === 'string' || Array.isArray(userCommand)) {
-				commands = [runSingleCommand(userCommand)];
-			} else {
-				commands = Object.keys(userCommand).map(name => {
-					const command = userCommand[name];
-					return runSingleCommand(command, name);
-				});
-			}
-			await Promise.all(commands);
-		} catch (err) {
-			infoOutput.event({
-				type: 'progress',
-				name: progressName,
-				status: 'failed',
+		let commands;
+		if (typeof userCommand === 'string' || Array.isArray(userCommand)) {
+			commands = [runSingleCommand(userCommand)];
+		} else {
+			commands = Object.keys(userCommand).map(name => {
+				const command = userCommand[name];
+				return runSingleCommand(command, name);
 			});
-			if (err && (err.code === 130 || err.signal === 2)) { // SIGINT seen on darwin as code === 130, would also make sense as signal === 2.
-				infoOutput.raw(`\r\n\x1b[1m${lifecycleHookName} interrupted.\x1b[0m\r\n\r\n`);
-			} else {
-				if (err?.code) {
-					infoOutput.write(toErrorText(`${lifecycleHookName} failed with exit code ${err.code}. Skipping any further user-provided commands.`));
-				}
-				throw new ContainerError({
-					description: `The ${lifecycleHookName} in the ${userCommandOrigin} failed.`,
-					originalError: err,
-				});
-			}
 		}
+		await Promise.all(commands);
```

Comment:
Keep this in a try-catch and move all infoOutput.event() calls here. These events don't seem to be per command but per lifecycle hook.